modules/system/oagw/docs/adr-storage-schema.md

```
@@ -0,0 +1,654 @@
+```sql
+SELECT u.* FROM oagw_upstream u
+JOIN (SELECT :t0 AS tid, 0 AS depth
+      UNION ALL SELECT :t1, 1
+      UNION ALL SELECT :t2, 2) h ON h.tid = u.tenant_id
+WHERE u.alias = :alias
+ORDER BY h.depth ASC;
+```
+
+**Application processing** — the result set is already sorted closest-first. The app takes the first row as the resolved upstream, but must still check ancestor rows: if any ancestor has `enabled = false`, the upstream is blocked.
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)